stl/inc/queue

```diff
@@ -313,6 +360,22 @@ template <class _Pr, class _Container, class _Alloc,
 priority_queue(_Pr, _Container, _Alloc) -> priority_queue<typename _Container::value_type, _Container, _Pr>;
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+template <class _Iter, class _Alloc, enable_if_t<conjunction_v<_Is_iterator<_Iter>, _Is_allocator<_Alloc>>, int> = 0>
+priority_queue(_Iter, _Iter, _Alloc)
+    -> priority_queue<_Iter_value_t<_Iter>, vector<_Iter_value_t<_Iter>, _Alloc>, less<_Iter_value_t<_Iter>>>;
+
+template <class _Iter, class Compare, class _Alloc,
+    enable_if_t<conjunction_v<_Is_iterator<_Iter>, _Is_allocator<_Alloc>>, int> = 0>
+priority_queue(_Iter, _Iter, Compare, _Alloc)
+    -> priority_queue<_Iter_value_t<_Iter>, vector<_Iter_value_t<_Iter>, _Alloc>, Compare>;
+
+template <class _Iter, class Compare, class Container, class _Alloc,
+    enable_if_t<conjunction_v<_Is_iterator<_Iter>, _Is_allocator<_Alloc>>, int> = 0>
+priority_queue(_Iter, _Iter, Compare, Container, _Alloc)

```

Comment:
According to https://eel.is/c++draft/container.adaptors.general#4.5 we need to constrain with `uses_allocator` because we have both `_Container` and `_Alloc` here.